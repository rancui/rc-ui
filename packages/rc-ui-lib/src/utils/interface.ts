import type { CSSProperties } from 'react';

export interface BaseTypeProps {
  style?: CSSProperties;
  className?: string;
  children?: React.ReactNode;
}

/** 指定挂载的节点 */
export type TeleportType = HTMLElement | (() => HTMLElement) | null;
/** 带有displayName的组件 */
export type WithDisplayNameReactElement = React.ReactElement & { type: { displayName: string } };
