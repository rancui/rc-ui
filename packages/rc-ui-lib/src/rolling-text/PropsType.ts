import { BaseTypeProps } from '../utils';

export type RollingTextDirection = 'up' | 'down';
export type RollingTextStopOrder = 'ltr' | 'rtl';

export interface RollingTextProps extends BaseTypeProps {
  /**
   * 数字高度，单位为 px
   * @default 40
   */
  height?: number;
  /**
   * 起始数值
   * @default 0
   */
  startNum?: number;
  /**
   * 目标数值
   */
  targetNum?: number;
  /**
   * 内容数组，用于翻转非数字内容
   * @default []
   */
  textList?: string[];
  /**
   * 动画时长，单位为秒
   * @default 2
   */
  duration?: number;
  /**
   * 是否自动开始动画
   * @default true
   */
  autoStart?: boolean;
  /**
   * 文本翻滚方向，值为 down 和 up
   * @default down
   */
  direction?: RollingTextDirection;
  /**
   * 各个数位动画停止先后顺序，值为 ltr 和 rtl
   * @default ltr
   */
  stopOrder?: RollingTextStopOrder;
}

export type RollingTextInstance = {
  start: () => void;
  reset: () => void;
};

export interface RollingTextItemProps {
  figureArr?: any[];
  /**
   * 起始数值
   * @default 0
   */
  delay?: number;
  /**
   * 动画时长，单位为秒
   * @default 2
   */
  duration?: number;
  /**
   * 是否自动开始动画
   * @default true
   */
  isStart: boolean;
  /**
   * 文本翻滚方向，值为 down 和 up
   * @default down
   */
  direction?: RollingTextDirection;
  /**
   * 数字高度，单位为 px
   * @default 40
   */
  height?: number;
}

export type RollingTextThemeVars = {
  rollingTextBackground?: string;
  rollingTextColor?: string;
  rollingTextFontSize?: string;
  rollingTextGap?: string;
  rollingTextItemWidth?: string;
  rollingTextItemBorderRadius?: string;
};
