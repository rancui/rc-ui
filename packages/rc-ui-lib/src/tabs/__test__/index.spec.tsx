/* eslint-disable @typescript-eslint/no-unused-expressions */
/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, cleanup, fireEvent, waitFor } from '@testing-library/react';
import TestsEvent from '../../../tests/events';
import { sleep } from '../../../tests/utils';
import Tabs from '..';
import TestElements from './testElement';
import { TabsInstance } from '../PropsType';

describe('Tabs', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('should emit click-tab event when tab is clicked', async () => {
    const onClickTab = jest.fn();

    const { container } = render(
      <Tabs onClickTab={onClickTab}>
        <Tabs.TabPane title="title1">1</Tabs.TabPane>
        <Tabs.TabPane title="title2">2</Tabs.TabPane>
      </Tabs>,
    );

    await sleep();
    const tabs = container.querySelector('.rc-tab');

    await fireEvent.click(tabs);
    expect(onClickTab).toHaveBeenCalledWith(
      expect.objectContaining({
        name: 0,
        title: 'title1',
        disabled: false,
      }),
    );
  });

  it('should not emit change event when TabPane is disabled and tab is clicked', async () => {
    const onChange = jest.fn();

    const { container } = render(
      <Tabs onChange={onChange}>
        <Tabs.TabPane title="title1">1</Tabs.TabPane>
        <Tabs.TabPane disabled title="title2">
          2
        </Tabs.TabPane>
        <Tabs.TabPane title="title3">3</Tabs.TabPane>
      </Tabs>,
    );

    await sleep();
    const tabs = container.querySelectorAll('.rc-tab');

    await fireEvent.click(tabs[1]);
    expect(onChange).toHaveBeenCalledTimes(0);
    await fireEvent.click(tabs[0]);
    expect(onChange).toHaveBeenCalledTimes(0);
    await fireEvent.click(tabs[2]);
    expect(onChange).toHaveBeenCalledTimes(1);
    await fireEvent.click(tabs[2]);
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('should match active tab by name when using name prop', async () => {
    const onClickTab = jest.fn();
    const { container } = render(
      <Tabs active="a" onClickTab={onClickTab}>
        <Tabs.TabPane title="title1" name="a">
          Tab
        </Tabs.TabPane>
        <Tabs.TabPane title="title2" name="b">
          Tab
        </Tabs.TabPane>
      </Tabs>,
    );
    const item = container.querySelectorAll('.rc-tab');
    await fireEvent.click(item[1]);
    expect(onClickTab).toHaveBeenCalledWith(
      expect.objectContaining({
        name: 'b',
        title: 'title2',
        disabled: false,
      }),
    );
  });

  it('should not render zero badge when show-zero-badge prop is false', async () => {
    const { container } = render(
      <Tabs>
        <Tabs.TabPane badge={0}>1</Tabs.TabPane>
        <Tabs.TabPane badge={0} showZeroBadge={false}>
          2
        </Tabs.TabPane>
      </Tabs>,
    );
    await sleep();
    expect(container).toMatchSnapshot();
  });

  it('should render card type correctly', async () => {
    const { container } = render(
      <Tabs active={0} type="card" color="#ff0000">
        <Tabs.TabPane badge={1}>1</Tabs.TabPane>
        <Tabs.TabPane dot>2</Tabs.TabPane>
        <Tabs.TabPane renderTitle={<h3>haha</h3>} />
        <Tabs.TabPane renderTitle={(active) => <h3>{active ? 'true' : 'false'}</h3>} />
      </Tabs>,
    );
    await sleep();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when lazy-render', async () => {
    const { container, rerender } = render(
      <Tabs active="a" lazyRender>
        <Tabs.TabPane title="title1" name="a">
          Tab
        </Tabs.TabPane>
        <Tabs.TabPane title="title2" name="b">
          Tab
        </Tabs.TabPane>
      </Tabs>,
    );
    await sleep();
    expect(container.querySelector('.rc-tabs__content')).toMatchSnapshot();

    rerender(
      <Tabs active="a" lazyRender={false}>
        <Tabs.TabPane title="title1" name="a">
          Tab
        </Tabs.TabPane>
        <Tabs.TabPane title="title2" name="b">
          Tab
        </Tabs.TabPane>
      </Tabs>,
    );

    expect(container.querySelector('.rc-tabs__content')).toMatchSnapshot();
  });

  it('should render correctly when set scrollspy prop', async () => {
    const onChange = jest.fn();
    Date.now = jest.fn(() => 1482363367071);
    const scrollspyConfig = {
      autoFocusLast: true,
      reachBottomThreshold: 50,
    };
    const { container } = render(
      <div className="test-rab">
        <Tabs sticky scrollspy={scrollspyConfig} onChange={onChange}>
          {[0, 1, 2, 3, 4].map((item) => (
            <Tabs.TabPane key={item} title={`title${item}`}>
              <div style={{ height: '80px' }}>内容 {item}</div>
            </Tabs.TabPane>
          ))}
        </Tabs>
      </div>,
    );
    const tabs = container.querySelectorAll('.rc-tab');
    await sleep();
    expect(container).toMatchSnapshot();
    fireEvent.click(tabs[2]);
    expect(onChange).toHaveBeenCalledWith(2, 'title2');
    expect(container).toMatchSnapshot();
  });
});

describe('Tabs test with testing library', () => {
  const mockRect = (el: Element, rect: Partial<DOMRect>) => {
    Object.defineProperty(el, 'getBoundingClientRect', {
      configurable: true,
      get: () => {
        return () => ({
          top: 0,
          left: 0,
          right: 0,
          bottom: 0,
          width: 0,
          height: 0,
          ...rect,
        });
      },
    });
  };

  const mockSticky = (c: Element, sticky: Element, rect?: Partial<DOMRect>) => {
    Object.defineProperty(c, 'offsetParent', {
      configurable: true,
      get: () => document.body,
    });

    if (rect) {
      mockRect(sticky, rect);
    }
  };

  const mockOffset = (c: Element, rect?: Partial<DOMRect>) => {
    Object.defineProperty(c, 'offsetHeight', {
      configurable: true,
      get: () => rect?.height,
    });
    Object.defineProperty(c, 'offsetWidth', {
      configurable: true,
      get: () => rect?.width,
    });
  };

  beforeEach(() => {
    window.scrollTo = jest.fn();
  });

  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('should scroll to specific swipe after calling the scrollTo method', async () => {
    const onChange = jest.fn();

    const tabsRef = React.createRef<TabsInstance>();
    render(
      <Tabs ref={tabsRef} onChange={onChange}>
        <Tabs.TabPane title="1">content1</Tabs.TabPane>
        <Tabs.TabPane disabled title="2">
          content2
        </Tabs.TabPane>
        <Tabs.TabPane title="3">content3</Tabs.TabPane>
      </Tabs>,
    );

    const scrollTo = (step: number) => {
      tabsRef.current.scrollTo(step);
    };

    await waitFor(async () => {
      scrollTo(1);
    });

    await sleep(100);

    expect(onChange).toHaveBeenCalledWith(2, '3');
  });
  it('should render correctly after inserting a tab', async () => {
    const { container, getByTestId } = render(<TestElements />);

    await sleep();
    expect(container).toMatchSnapshot();

    fireEvent.click(getByTestId('button-change'));

    await sleep();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly when set sticky prop', async () => {
    Object.defineProperty(window.HTMLElement.prototype, 'clientHeight', {
      value: 300,
    });
    const onScroll = jest.fn();
    const { container } = render(
      <Tabs
        sticky
        lineHeight={5}
        scrollspy={{ autoFocusLast: true, reachBottomThreshold: 100 }}
        onScroll={onScroll}
      >
        <Tabs.TabPane title="1">content1</Tabs.TabPane>
        <Tabs.TabPane title="2" disabled>
          content2
        </Tabs.TabPane>
        <Tabs.TabPane title="3">content3</Tabs.TabPane>
      </Tabs>,
    );

    await sleep();

    const sticky = container.querySelector('.rc-sticky');
    const c = container.querySelector('.rc-tabs');

    c && mockRect(c, { width: 300, height: 200, top: 0, bottom: 600 });
    sticky &&
      mockSticky(c, sticky, {
        width: 300,
        height: 50,
        top: -100,
        bottom: -90,
      });
    Object.defineProperty(sticky.parentNode, 'offsetParent', {
      configurable: true,
      get: () => document.body,
    });

    await waitFor(() => {
      fireEvent.scroll(window, { target: { pageYOffset: 100 } }); // 0
    });

    expect(container).toMatchSnapshot();
  });

  it('should swipe to switch tab', async () => {
    const onChange = jest.fn();
    Date.now = jest.fn(() => 1482363367071);
    const { container } = render(
      <Tabs sticky active="b" swipeable onChange={onChange}>
        <Tabs.TabPane title="title1" name="a">
          Tab
        </Tabs.TabPane>
        <Tabs.TabPane title="title2" name="b">
          Tab
        </Tabs.TabPane>
        <Tabs.TabPane title="title3" name="c">
          Tab
        </Tabs.TabPane>
      </Tabs>,
    );

    const sticky = container.querySelector('.rc-sticky');
    const c = container.querySelector('.rc-tabs');
    const track = container.querySelector('.rc-swiper__track');

    c && mockRect(c, { width: 300, height: 200, top: 0, bottom: 600 });
    sticky &&
      mockSticky(c, sticky, {
        width: 300,
        height: 50,
        top: -10,
      });

    mockOffset(track, {
      width: 300,
      height: 50,
    });

    await TestsEvent.triggerDrag(track, [-300, 0]);
    await sleep(300);

    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
