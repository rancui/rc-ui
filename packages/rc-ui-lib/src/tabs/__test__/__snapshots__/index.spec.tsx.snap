// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should not render zero badge when show-zero-badge prop is false 1`] = `
<div>
  <div
    class="rc-tabs rc-tabs--line"
  >
    <div
      class="rc-tabs__wrap"
    >
      <div
        class="rc-tabs__nav rc-tabs__nav--line"
        role="tablist"
      >
        <div
          aria-selected="true"
          class="rc-tab rc-tab--active"
        >
          <div
            class="rc-badge__wrapper"
          >
            <span
              class="rc-tab__text rc-tab__text--ellipsis"
            />
            <div
              class="rc-badge rc-badge--top-right rc-badge--fixed"
            >
              0
            </div>
          </div>
        </div>
        <div
          aria-selected="false"
          class="rc-tab"
        >
          <div
            class="rc-badge__wrapper"
          >
            <span
              class="rc-tab__text rc-tab__text--ellipsis"
            />
          </div>
        </div>
        <div
          class="rc-tabs__line"
        />
      </div>
    </div>
    <div
      class="rc-tabs__content"
    >
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        1
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: none;"
      />
    </div>
  </div>
</div>
`;

exports[`Tabs should render card type correctly 1`] = `
<div>
  <div
    class="rc-tabs rc-tabs--card"
  >
    <div
      class="rc-tabs__wrap"
    >
      <div
        class="rc-tabs__nav rc-tabs__nav--card"
        role="tablist"
        style="border-color: #ff0000;"
      >
        <div
          aria-selected="true"
          class="rc-tab rc-tab--active"
          style="border-color: #ff0000; background-color: rgb(255, 0, 0);"
        >
          <div
            class="rc-badge__wrapper"
          >
            <span
              class="rc-tab__text rc-tab__text--ellipsis"
            />
            <div
              class="rc-badge rc-badge--top-right rc-badge--fixed"
            >
              1
            </div>
          </div>
        </div>
        <div
          aria-selected="false"
          class="rc-tab"
          style="border-color: #ff0000; color: rgb(255, 0, 0);"
        >
          <div
            class="rc-badge__wrapper"
          >
            <span
              class="rc-tab__text rc-tab__text--ellipsis"
            />
            <div
              class="rc-badge rc-badge--top-right rc-badge--dot rc-badge--fixed"
            />
          </div>
        </div>
        <div
          aria-selected="false"
          class="rc-tab"
          style="border-color: #ff0000; color: rgb(255, 0, 0);"
        >
          <span
            class="rc-tab__text rc-tab__text--ellipsis"
          >
            <h3>
              haha
            </h3>
          </span>
        </div>
        <div
          aria-selected="false"
          class="rc-tab"
          style="border-color: #ff0000; color: rgb(255, 0, 0);"
        >
          <span
            class="rc-tab__text rc-tab__text--ellipsis"
          >
            <h3>
              false
            </h3>
          </span>
        </div>
      </div>
    </div>
    <div
      class="rc-tabs__content"
    >
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        1
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: none;"
      />
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: none;"
      />
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: none;"
      />
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly when lazy-render 1`] = `
<div
  class="rc-tabs__content"
>
  <div
    class="rc-tab__pane"
    role="tabpanel"
    style="display: block;"
  >
    Tab
  </div>
  <div
    class="rc-tab__pane"
    role="tabpanel"
    style="display: none;"
  />
</div>
`;

exports[`Tabs should render correctly when lazy-render 2`] = `
<div
  class="rc-tabs__content"
>
  <div
    class="rc-tab__pane"
    role="tabpanel"
    style="display: block;"
  >
    Tab
  </div>
  <div
    class="rc-tab__pane"
    role="tabpanel"
    style="display: none;"
  >
    Tab
  </div>
</div>
`;

exports[`Tabs should render correctly when set scrollspy prop 1`] = `
<div>
  <div
    class="test-rab"
  >
    <div
      class="rc-tabs rc-tabs--line"
    >
      <div>
        <div
          class="rc-sticky"
        >
          <div
            class="rc-tabs__wrap"
          >
            <div
              class="rc-tabs__nav rc-tabs__nav--line"
              role="tablist"
            >
              <div
                aria-selected="true"
                class="rc-tab rc-tab--active"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title0
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title1
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title2
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title3
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title4
                </span>
              </div>
              <div
                class="rc-tabs__line"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="rc-tabs__content"
      >
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            0
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            1
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            2
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            3
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly when set scrollspy prop 2`] = `
<div>
  <div
    class="test-rab"
  >
    <div
      class="rc-tabs rc-tabs--line"
    >
      <div>
        <div
          class="rc-sticky"
        >
          <div
            class="rc-tabs__wrap"
          >
            <div
              class="rc-tabs__nav rc-tabs__nav--line"
              role="tablist"
            >
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title0
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title1
                </span>
              </div>
              <div
                aria-selected="true"
                class="rc-tab rc-tab--active"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title2
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title3
                </span>
              </div>
              <div
                aria-selected="false"
                class="rc-tab"
              >
                <span
                  class="rc-tab__text rc-tab__text--ellipsis"
                >
                  title4
                </span>
              </div>
              <div
                class="rc-tabs__line"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="rc-tabs__content"
      >
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            0
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            1
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            2
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            3
          </div>
        </div>
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          <div
            style="height: 80px;"
          >
            内容 
            4
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs test with testing library should render correctly after inserting a tab 1`] = `
<div>
  <div
    class="rc-tabs rc-tabs--line"
  >
    <div>
      <div
        class="rc-sticky"
      >
        <div
          class="rc-tabs__wrap"
        >
          <div
            class="rc-tabs__nav rc-tabs__nav--line"
            role="tablist"
          >
            <div
              aria-selected="true"
              class="rc-tab rc-tab--active"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                title1
              </span>
            </div>
            <div
              aria-selected="false"
              class="rc-tab"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                title3
              </span>
            </div>
            <div
              class="rc-tabs__line"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="rc-tabs__content"
    >
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        Tab
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        Tab
      </div>
    </div>
  </div>
  <button
    data-testid="button-change"
    type="button"
  >
    ChangeInsert
  </button>
</div>
`;

exports[`Tabs test with testing library should render correctly after inserting a tab 2`] = `
<div>
  <div
    class="rc-tabs rc-tabs--line"
  >
    <div>
      <div
        class="rc-sticky"
      >
        <div
          class="rc-tabs__wrap"
        >
          <div
            class="rc-tabs__nav rc-tabs__nav--line"
            role="tablist"
          >
            <div
              aria-selected="true"
              class="rc-tab rc-tab--active"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                title1
              </span>
            </div>
            <div
              aria-selected="false"
              class="rc-tab"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              />
            </div>
            <div
              aria-selected="false"
              class="rc-tab"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                title3
              </span>
            </div>
            <div
              class="rc-tabs__line"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="rc-tabs__content"
    >
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        Tab
      </div>
      <div
        index="1"
      >
        <div
          class="rc-tab__pane"
          role="tabpanel"
          style="display: block;"
        >
          Tab
        </div>
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        Tab
      </div>
    </div>
  </div>
  <button
    data-testid="button-change"
    type="button"
  >
    ChangeInsert
  </button>
</div>
`;

exports[`Tabs test with testing library should render correctly when set sticky prop 1`] = `
<div>
  <div
    class="rc-tabs rc-tabs--line"
  >
    <div>
      <div
        class="rc-sticky"
      >
        <div
          class="rc-tabs__wrap"
        >
          <div
            class="rc-tabs__nav rc-tabs__nav--line"
            role="tablist"
          >
            <div
              aria-selected="false"
              class="rc-tab"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                1
              </span>
            </div>
            <div
              aria-selected="false"
              class="rc-tab rc-tab--disabled"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                2
              </span>
            </div>
            <div
              aria-selected="true"
              class="rc-tab rc-tab--active"
            >
              <span
                class="rc-tab__text rc-tab__text--ellipsis"
              >
                3
              </span>
            </div>
            <div
              class="rc-tabs__line"
              style="transform: translateX(0px) translateX(-50%); transition-duration: 300ms; height: 5px; border-radius: 5px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="rc-tabs__content"
    >
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        content1
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        content2
      </div>
      <div
        class="rc-tab__pane"
        role="tabpanel"
        style="display: block;"
      >
        content3
      </div>
    </div>
  </div>
</div>
`;
